src/ProfileEndpoints.jl

```diff
@@ -94,7 +94,8 @@ function cpu_profile_stop_endpoint(req::HTTP.Request)
     qp = HTTP.queryparams(uri)
     with_pprof = parse(Bool, get(qp, "pprof", default_pprof()))
     filename = "cpu_profile"
-    return _cpu_profile_response(filename; with_pprof)
+    # Defer the potentially expensive profile symbolication to a non-interactive thread
+    return fetch(Threads.@spawn _cpu_profile_response($filename; $with_pprof))
```

Comment:
Can you do line 107 too?

And the other endpoints need to be done as well. If you want to tackle them in this PR, that would be awesome, otherwise please file an issue and we can get to it later. Thanks!